idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/markers/valueArgumentCoercion.kt

```diff
@@ -0,0 +1,54 @@
+package arrow.meta.ide.plugins.proofs.markers
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.plugins.proofs.inspections.PairTypes
+import arrow.meta.ide.plugins.proofs.inspections.PairTypes.Companion.pairOrNull
+import arrow.meta.ide.plugins.proofs.inspections.resolveKotlinType
+import arrow.meta.ide.resources.ArrowIcons
+import arrow.meta.phases.CompilerContext
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.plugins.proofs.phases.areTypesCoerced
+import arrow.meta.plugins.proofs.phases.coerceProof
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtValueArgument
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+val IdeMetaPlugin.implicitCoercionValueArgumentLineMarker: ExtensionPhase
+  get() = addLineMarkerProviderM(
+    icon = ArrowIcons.ICON4,
+    composite = KtValueArgument::class.java,
+    transform = { psiElement: PsiElement ->
+      psiElement.ctx()?.let { ctx ->
+        psiElement.safeAs<KtValueArgument>()?.takeIf {
+          it.isCoerced(ctx)
+        }
+      }
+    },
+    message = { ktElement: KtValueArgument ->
+      ktElement.participatingTypes().mapNotNull { (subtype, supertype) ->
```

Comment:
Same for this block, even if we end up printing a different message.

Reply:
this I did not understand.. wanna avoid duplication calling participatingTypes ?

Reply:
Explained above in a previous comment, let me know if you wanna have a call 🙏